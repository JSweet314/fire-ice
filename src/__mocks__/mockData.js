/* eslint-disable max-len*/
export const mockHouses = [{
  ancestralWeapons: ["Lady Forlorn"],
  cadetBranches: [],
  coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
  currentLord: "https://www.anapioficeandfire.com/api/characters/681",
  diedOut: "",
  founded: "",
  founder: "https://www.anapioficeandfire.com/api/characters/256",
  heir: "https://www.anapioficeandfire.com/api/characters/677",
  name: "House Corbray of Heart's Home",
  overlord: "https://www.anapioficeandfire.com/api/houses/7",
  region: "",
  seats: ["Heart's Home"],
  swornMembers: [
    "https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"
  ],
  titles: ["Lord of Heart's Home", "King of the Fingers (historical)"],
  url: "https://www.anapioficeandfire.com/api/houses/81",
  words: ""
}];

export const mockCleanHouses = [{ "ancestralWeapons": "Lady Forlorn", "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)", "founded": "", "name": "House Corbray of Heart's Home", "seats": "Heart's Home", "swornMembers": ["255", "256", "447", "654", "677", "681", "871", "1182"], "titles": ["Lord of Heart's Home", "King of the Fingers (historical)"], "words": "" }];

export const mockCharacter = {
  "url": "https://www.anapioficeandfire.com/api/characters/255",
  "name": "Corwyn Corbray",
  "gender": "Male",
  "culture": "",
  "born": "",
  "died": "In 134 AC, at Runestone",
  "titles": [
    "Ser",
    "Lord Regent"
  ],
  "aliases": [
    ""
  ],
  "father": "",
  "mother": "",
  "spouse": "https://www.anapioficeandfire.com/api/characters/871",
  "allegiances": [
    "https://www.anapioficeandfire.com/api/houses/81"
  ],
  "books": [
    "https://www.anapioficeandfire.com/api/books/11"
  ],
  "povBooks": [],
  "tvSeries": [
    ""
  ],
  "playedBy": [
    ""
  ]
};

export const mockCleanCharacter = { "died": "In 134 AC, at Runestone", "name": "Corwyn Corbray" };