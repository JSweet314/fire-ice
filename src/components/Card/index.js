import React from 'react';

const Card = ({name, founded, seats, titles, coatOfArms, ancestralWeapons, words}) => {
  const titlePs = titles.map(title => <p key={title}>Title: {title}</p>);
  return (
    <article className='Card'>
      <h2>{name}</h2>
      <h3>{words}</h3>
      <p>Seats: {seats}</p>
      {titlePs}
      <p>Ancestral Weapons: {ancestralWeapons}</p>
      <p>Coat of Arms: {coatOfArms}</p>
    </article>
  )
};

export default Card;