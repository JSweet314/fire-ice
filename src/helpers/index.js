export const housesDataWrangler = housesArray => 
  housesArray.map(house => ({
    name: house.name,
    founded: house.founded,
    seats: house.seats.join(', '),
    titles: house.titles,
    coatOfArms: house.coatOfArms,
    ancestralWeapons: house.ancestralWeapons.join(', '),
    words: house.words
  }));