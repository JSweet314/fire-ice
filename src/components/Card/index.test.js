import React from 'react';
import { shallow } from 'enzyme';
import Card from './index';
import { housesDataWrangler } from '../../helpers';
import { mockHouses } from '../../__mocks__/mockData';

describe('Card', () => {
  const mockHouse = housesDataWrangler(mockHouses)[0];
  it('should match a snapshot', () => {
    const wrapper = shallow(<Card {...mockHouse} />);
    expect(wrapper).toMatchSnapshot();
  });
});