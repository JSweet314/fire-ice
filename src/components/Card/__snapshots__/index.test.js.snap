// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    ancestralWeapons="Lady Forlorn"
    coatOfArms="Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)"
    founded=""
    name="House Corbray of Heart's Home"
    seats="Heart's Home"
    titles={
        Array [
            "Lord of Heart's Home",
            "King of the Fingers (historical)",
          ]
    }
    words=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          House Corbray of Heart's Home
</h2>,
        <h3 />,
        <p>
          Seats: 
          Heart's Home
</p>,
        Array [
          <p>
            Title: 
            Lord of Heart's Home
</p>,
          <p>
            Title: 
            King of the Fingers (historical)
</p>,
        ],
        <p>
          Ancestral Weapons: 
          Lady Forlorn
</p>,
        <p>
          Coat of Arms: 
          Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</p>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "House Corbray of Heart's Home",
        },
        "ref": null,
        "rendered": "House Corbray of Heart's Home",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Seats: ",
            "Heart's Home",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Seats: ",
          "Heart's Home",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Lord of Heart's Home",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Title: ",
            "Lord of Heart's Home",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Title: ",
          "Lord of Heart's Home",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "King of the Fingers (historical)",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Title: ",
            "King of the Fingers (historical)",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Title: ",
          "King of the Fingers (historical)",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ancestral Weapons: ",
            "Lady Forlorn",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Ancestral Weapons: ",
          "Lady Forlorn",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Coat of Arms: ",
            "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Coat of Arms: ",
          "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            House Corbray of Heart's Home
</h2>,
          <h3 />,
          <p>
            Seats: 
            Heart's Home
</p>,
          Array [
            <p>
              Title: 
              Lord of Heart's Home
</p>,
            <p>
              Title: 
              King of the Fingers (historical)
</p>,
          ],
          <p>
            Ancestral Weapons: 
            Lady Forlorn
</p>,
          <p>
            Coat of Arms: 
            Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</p>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seats: ",
              "Heart's Home",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Seats: ",
            "Heart's Home",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Lord of Heart's Home",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              "Lord of Heart's Home",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            "Lord of Heart's Home",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "King of the Fingers (historical)",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              "King of the Fingers (historical)",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            "King of the Fingers (historical)",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ancestral Weapons: ",
              "Lady Forlorn",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Ancestral Weapons: ",
            "Lady Forlorn",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat of Arms: ",
              "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Coat of Arms: ",
            "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
